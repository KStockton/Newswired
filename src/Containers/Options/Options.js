import React, { Component } from 'react';
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { 
  faBookOpen, 
  faFire, 
  faRunning, 
  faGlobeAfrica, 
  faCheck, 
  faArrowRight } from '@fortawesome/free-solid-svg-icons';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import { setCategories } from '../../actions/index';
import PropTypes from 'prop-types';

export class Options extends Component {

  
  selectTopic = (event) => {
    const { name } = event.target;
    this.props.setCategories(name);
  }
  
  render() {
    let numOfCategories = Object.values(this.props.categories).filter(item=> item === true).length;

    const { topNews, books, sportsNews, travel } = this.props.categories;

    return (
      <section>
        <div className="Options-background">
          <div className="Options-header">
            <h1 className="Options-title">News Wired</h1>
          </div>
          <header className="Options-news">
            <h4 className="Options-select-choice">Select News</h4>
          </header>
          <div className="Options-wrapper">
            <div className="Options-btn-wrapper">
              <button className="Options-btn" name="topNews" onClick={(event) => this.selectTopic(event)}>
                <label htmlFor="top-news">
                  <FontAwesomeIcon id="top-news" className="Options-fontawesome select-options" icon={faFire}/>
                Top News
                </label>
              </button>
              { topNews ? 
                <FontAwesomeIcon  className="Options-check" icon={faCheck}/> : <FontAwesomeIcon  className="Options-uncheck" icon={faCheck}/> }
            </div>
            <div className="Options-btn-wrapper">
              <button className="Options-btn" name="books" onClick={this.selectTopic}>
                <label htmlFor="new-york-times-books">
                  <FontAwesomeIcon id="new-york-times-books" className="Options-fontawesome select-options" icon={faBookOpen}/>
                  NYT Best Sellers
                </label>
              </button>
              { books ?  <FontAwesomeIcon  className="Options-check" icon={faCheck}/> : <FontAwesomeIcon  className="Options-uncheck" icon={faCheck}/>}
            </div>
            <div className="Options-btn-wrapper">
              <button className="Options-btn"  name="sportsNews" onClick={this.selectTopic}>
                <label htmlFor="sports-news">
                  <FontAwesomeIcon id="sports-news" className="Options-fontawesome select-options" icon={faRunning}/>
                  Sports 
                </label>
              </button>
              { sportsNews ? <FontAwesomeIcon  className="Options-check select-options" icon={faCheck}/> : <FontAwesomeIcon  className="Options-uncheck" icon={faCheck}/>}
            </div>
            <div className="Options-btn-wrapper">
              <button className="Options-btn" name="travel" onClick={this.selectTopic}>
                <label htmlFor="latest-news">
                  <FontAwesomeIcon id="latest-news" className="Options-fontawesome select-options" icon={faGlobeAfrica}/>
                  Travel
                </label>
              </button>
              { travel ? <FontAwesomeIcon  className="Options-check" icon={faCheck}/> : <FontAwesomeIcon  className="Options-uncheck" icon={faCheck}/> }
            </div>
          </div>
          <div className="Options-continue" role="button">
            { 
              numOfCategories >= 1 && 
              <div id='next-container'>
                <FontAwesomeIcon id="latest-news" className="Options-fontawesome" icon={faArrowRight}/>
                <NavLink to="/Main" id="next"> 
                  <p>continue</p>
                </NavLink>
              </div>
            }
          </div>
        </div>
      </section>
    );
  }
}

export const mapStateToProps = (state) => ({
  categories: state.categories
});

export const mapDispatchToProps = (dispatch) => ({
  setCategories: (category) => dispatch(setCategories(category))
});

Options.propTypes = {
  categories: PropTypes.object,
  setCategories: PropTypes.func
};

export default connect(mapStateToProps, mapDispatchToProps)(Options);
