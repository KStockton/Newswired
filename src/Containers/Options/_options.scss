@import '../../Sass/_variables';

.Options-background {
  background-image: $splash;
}
.Options-header {
  background-color: #ccd0d56b;
  margin: 0;
  width: 100vw;
  height: 123px;
  display: flex;
  align-items: center;
  justify-content: center;
  
  .Options-title {
    margin: 0;
    font-family: 'Old Standard TT', serif;
    text-align: center;
    font-size: 3rem;
    padding-top: 8px;
  }
}

.Options-news {
  margin: 0 auto;
  padding-top: 100px;
  padding-bottom: 50px;
}
.Options-select {
  font-family: 'Noto Sans', sans-serif;
  font-weight: 400;
  font-size: 40px;
}

.Options-wrapper {
  background-size: 100% 100%;
  width: 100vw;
  height: 87vh;
  display: flex;
  justify-content: space-evenly;
}

.Options-btn {
  padding: 10px;
  position: relative;
  text-decoration: none;
  color: #000;
  border: solid 1px #186f8f;
  border-radius: 5px;
  text-transform: uppercase;
  z-index: auto;
  box-shadow: 3px 3px 2px #837a7a;
  height: 70px;
  width: 224px;
  font-size: 0.8rem;
  display: flex;
  justify-content: space-evenly;
}
.Options-btn:hover {
  padding-left: 10px;
    padding-right: 10px;
    padding-top: 11px;
    background-image: -webkit-gradient(linear, left bottom, left top, color-stop(0, #2cffc9), color-stop(1, #ffffff));
    color: white;
    background: black;
}

.Options-btn::before {
  background-color: #ccd0d56b;
  content: "";
  display: block;
  position: absolute;
  width: 100%;
  height: 100%;
  padding: 8px;
  left: -8px;
  top: -8px;
  border-radius: 5px;
  box-shadow: inset 0px 1px 1px #909193, 0px 1px 0px #fff;
}

.Options-btn::active {
  padding-bottom: 9px;
  padding-left: 10px;
  padding-right: 10px;
  padding-top: 11px;
  top: 1px;
  background-image: -webkit-gradient( linear, left bottom, left top, color-stop(0, rgb(62,184,229)), color-stop(1, rgb(44,160,202)) );
}

.Options-fontawesome {
  padding-right: 9px;
  font-size: 23px;
}

.Options-btn-wrapper {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.Options-check {
  min-height: 34px;
  margin: 2rem;
  font-size: 1.5rem;
  color: green;
}
#next{
  width: 100px;
  height: 49px;
  display: flex;
  align-items: center;
}

.Options-continue {
  position: fixed;
  top: 61%;
  left: 47%;
  margin: 0 auto;
}