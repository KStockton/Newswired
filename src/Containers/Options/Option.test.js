import React from 'react'
import {Option} from './Options'
import { shallow } from 'enzyme'


describe('Option', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Option />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})