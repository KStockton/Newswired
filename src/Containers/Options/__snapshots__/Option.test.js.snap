// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Option should match the snapshot 1`] = `undefined`;
